db-tools-ij/src/main/java/com/splicemachine/db/impl/tools/ij/ProgressThread.java

```diff
@@ -0,0 +1,140 @@
+    private String progressBar =  "[------------+----------25%----------+-----------50%-----------+----------75%-----------+-----------]";
+    private byte[] progressBarB = progressBar.getBytes();
+    private int PROGRESS_BAR_LENGTH = progressBarB.length;
+
+    Supplier<String> progressInfoProvider;
+    PrintStream outputStream;
+    boolean started = false;
+    boolean hadException = false;
+    public boolean canceled = false;
+
+    // used in tests only
+    public ProgressThread(Supplier<String> op, PrintStream outputStream) {
+        progressInfoProvider = op;
+        this.outputStream = outputStream;
+    }
+
+    public ProgressThread(ij ijParser, String command, PrintStream outputStream) {
+        progressInfoProvider = () -> {
+            try {
+                return ijParser.getProgress(command);
+            } catch (SQLException e) {
+                return "";
+            }
+        };
+        this.outputStream = outputStream;
+    }
+
+    @Override
+    public void start() {
+        started = true;
+        super.start();
+    }
+
+    private void printProgressBarChar(int i) {
+        outputStream.print( (char)progressBarB[i] );
+    }
+
+    public void finishCurrentProgressBar() {
+        if(!startedProgress) return;
+        for(int i = currentPercentage; i< PROGRESS_BAR_LENGTH; i++) {
+            printProgressBarChar(i);
+        }
+        outputStream.print("\n");
+    }
+
+    private void updateProgress(ProgressInfo pi) {
+        startedProgress = true;
+
+        if( currentJob != pi.getJobNumber() || currentStage != pi.getNumCompletedStages()) {
+            // JOB or STAGE has changed
+            if(currentJob != -1)
+                finishCurrentProgressBar();
+            currentJob = pi.getJobNumber();
+            currentStage = pi.getNumCompletedStages();
+            currentPercentage = 0;
+            outputStream.println(pi.getJobname() + " (Job " + (pi.getJobNumber()+1) + ", Stage "
+                    + (pi.getNumCompletedStages()+1) + " of " + pi.getNumStages() + ")");
+        }
+        int perc = (pi.getNumCompletedTasks()* PROGRESS_BAR_LENGTH)/pi.getNumTasks();
+        for(int i=currentPercentage; i<perc; i++) {
+            printProgressBarChar(i);
+        }
+        currentPercentage = perc;
+    }
+
+    public void updateProgress(String s) {
+        if( s.length() > 0 && !s.equals("-") ) {
+            try {
+                updateProgress(ProgressInfo.deserializeFromString(s));
+            } catch (Exception e) {
+                if(!hadException) {
+                    outputStream.println("\nException while parsing/updating ProgressInfo: " + e.toString());
+                    hadException = true; // ignore following to not spam
+                }
+            }
+        }
+    }
+
+    public void run() {
+        outputStream.println(firstRunning);
```

Comment:
Would it be possible to print this / start this thread only for Spark queries?

Reply:
currently not